src/main/scala/no/ndla/articleapi/controller/ArticleController.scala

```diff
@@ -84,10 +124,36 @@ trait ArticleController {
 
       readService.withId(articleId) match {
         case Some(image) => image
-        case None => halt(status = 404, body = Error(Error.NOT_FOUND, s"No article with id $articleId found"))
+        case None => NotFound(body = Error(Error.NOT_FOUND, s"No article with id $articleId found"))
       }
+    }
 
+    post("/", operation(newArticle)) {
+      val newArticle = extract[NewArticle](request.body)
+      val article = writeService.newArticle(newArticle)
+      logger.info(s"CREATED article with ID = ${article.id}")
+      Created(body=article)
+    }
+
+    patch("/:article_id", operation(updateArticle)) {
+      val articleId = long("article_id")
+      val updatedArticle = extract[UpdatedArticle](request.body)
+      val article = writeService.updateArticle(articleId, updatedArticle).get
```

Comment:
Denne er jeg ingen fan av. Her kjøres en `get`på Try-en som vi får fra writeService. Om den returnerer en Failure så kastes exceptionen. og forøvrig catches av errorhandleren, men her burde vi heller ha en matching som gjør noe mer fornuftig. For å unngå å måtte skrive feilhåndtering både her og i errorHandler (i NdlaController), så burde vi trekke den ut i en funksjon, som vi kan kalle herfra.
F.eks:

```
writeService.updateArticle(articleId, updatedArticle) match {
        case Success(article) => Ok(body=article)
        case Failure(e: Exception) => errorHandler(e)
}
```
Loggingen om at en artikkel har blitt oppdatert, synes jeg også skal skje i writeService.